slackify-html.js

```diff
@@ -38,6 +38,96 @@ function walkList(dom, ordered, nesting, start) {
             // Add the text content of the p element
             out += walkList(el.children, ordered, nesting) + "\n";
             break;
+          case "a":
+            if (el.attribs && el.attribs.href) {
+              out += "<" + el.attribs.href + "|" + walkList(el.children) + "> ";
+            } else {
+              out += walkList(el.children);
+            }
+            break;
+          case "code":
+            out += "`" + walkList(el.children) + "` ";
+            break;
+          case "del":
+            out += "~" + walkList(el.children) + "~ ";
+            break;
+          case "strong":
+            content = walkList(el.children);
+            var contentArr = content.split("\n");
+            var innerOutput = "";
+            for (var i = 0; i < contentArr.length; i++) {
+              content = contentArr[i];
+              if (content.trim() !== "") {
+                var prefixSpace = false;
+                var suffixSpace = false;
+                if (content && content.charAt(0) === " ") {
+                  content = content.substr(1, content.length);
+                  prefixSpace = true;
+                }
+                if (content && content.charAt(content.length - 1) === " ") {
+                  content = content.substr(0, content.length - 1);
+                  suffixSpace = true;
+                }
+                if (prefixSpace) {
+                  innerOutput += " ";
+                }
+                if (
+                  el.name === "h1" ||
+                  el.name === "h2" ||
+                  el.name === "h3" ||
+                  el.name === "h4"
+                ) {
+                  content = content.replace(/\*/g, "");
```

Comment:
is it possible that the header itself will have the * character? like what if the header was something like `<h1>look at this sh*t</h1>`? would this line get rid of that character and make it `<h1>look at this sht</h1>` instread?

Reply:
I think I can actually get rid of this conditional because isn't it impossible to have these header tags in the list item?

Reply:
ah yeah, I think that's correct!